src/main/java/org/folio/edge/ncip/NcipHandler.java

```diff
@@ -1,156 +1,117 @@
-			          final NcipOkapiClient ncipOkapiClient = new NcipOkapiClient(client);
-				      action.apply(ncipOkapiClient, params);
-			    });
-		  }
-		  
-		  
-		  protected void handleConfigCheck(RoutingContext ctx) {
-			  
-			    handleCommon(ctx,
-			        new String[] {},
-			        new String[] {},
-			        (client, params) -> {
-			          logger.info("sending GET request to NcipOkapiClient");
-			          ((NcipOkapiClient) client).ncipConfigCheck(
-			              ctx.getBodyAsString(),
-			              ctx.request().headers(),
-			              resp -> handleProxyResponse(ctx, resp),
-			              t -> handleProxyException(ctx, t));
-			        });
-		  }
-		  
-		  protected void handleHealthCheck(RoutingContext ctx) {
-			  
-			    handleCommon(ctx,
-			        new String[] {},
-			        new String[] {},
-			        (client, params) -> {
-			          logger.info("sending GET request to NcipOkapiClient - health check");
-			          ((NcipOkapiClient) client).ncipHealthCheck(
-			              ctx.getBodyAsString(),
-			              ctx.request().headers(),
-			              resp -> handleProxyResponse(ctx, resp),
-			              t -> handleProxyException(ctx, t));
-			        });
-		  }
-		  
-		  
-
-		  protected void handle(RoutingContext ctx) {
-			  
-			    handleCommon(ctx,
-			        new String[] {},
-			        new String[] {},
-			        (client, params) -> {
-			          logger.info("sending POST request to NcipOkapiClient");
-			          ((NcipOkapiClient) client).callNcip(
-			              ctx.getBodyAsString(),
-			              ctx.request().headers(),
-			              resp -> handleProxyResponse(ctx, resp),
-			              t -> handleProxyException(ctx, t));
-			        });
-		  }
-		  
-		  @Override
-		  protected void handleProxyException(RoutingContext ctx, Throwable t) {
-		    logger.error("Exception calling OKAPI", t);
-		    if (t instanceof TimeoutException) {
-		      requestTimeout(ctx, t.getMessage());
-		    } else {
-		      internalServerError(ctx, t.getMessage());
-		    }
-		  }
-		  
-		  @Override
-		  protected void internalServerError(RoutingContext ctx, String msg) {
-		    if (!ctx.response().ended()) {
-		      ctx.response()
-		        .setStatusCode(500)
-		        .putHeader(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON)
-		        .end(getStructuredErrorMessage(500, MSG_INTERNAL_SERVER_ERROR));
-		    }
-		  }
-		  
-		  private String getStructuredErrorMessage(int statusCode, String message){
-			    String finalMsg;
-			    try{
-			      ErrorMessage error = new ErrorMessage(statusCode, message);
-			      finalMsg = error.toJson();
-			    }
-			    catch(JsonProcessingException ex){
-			      finalMsg = "{ code : \"\", message : \"" + message + "\" }";
-			    }
-			    return finalMsg;
-		}
-
-		  protected void handleProxyResponse(RoutingContext ctx, HttpClientResponse resp) {
```

Comment:
It is not used so, it is deleted

Reply:
I see that it is used in handleHealthCheck()

Reply:
Thanks, reverting

Reply:
Yes it was not used, instead org.folio.edge.core.Handler::handleProxyResponse is used